<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RouteEditor</class>
 <widget class="QDialog" name="RouteEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>740</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>740</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Route Editor</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_3" columnstretch="2,1">
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Inbound</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QPushButton" name="addInboundBtn">
          <property name="toolTip">
           <string>Add outbound</string>
          </property>
          <property name="text">
           <string>Add Inbound</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addDefaultBtn">
          <property name="toolTip">
           <string>Add default inbound from global config</string>
          </property>
          <property name="text">
           <string>Add Global Settings</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Outbound</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>New Outbound</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addOutboundBtn">
            <property name="toolTip">
             <string>Add outbound</string>
            </property>
            <property name="text">
             <string>Import</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="insertBlackBtn">
            <property name="toolTip">
             <string>Add blackhole outbound</string>
            </property>
            <property name="text">
             <string>Blackhole</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="insertDirectBtn">
            <property name="toolTip">
             <string>Add Freedom outbound</string>
            </property>
            <property name="text">
             <string>Direct / Freedom</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Existing Outbound</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="importGroupBtn"/>
          </item>
          <item>
           <widget class="QComboBox" name="importConnBtn"/>
          </item>
          <item>
           <widget class="QToolButton" name="importExistingBtn">
            <property name="text">
             <string>OK</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="title">
        <string>Selected Inbound/Outbound Info</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="2">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Protocol</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="tagLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Tag</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Host</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="protocolLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="hostLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_19">
          <property name="text">
           <string>Port</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="portLabel">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Routes</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1,0">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>A rule with no inbound connected means there's no inbound restriction.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="nodeGraphWidget" native="true">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>150</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="routesControlButtons">
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="editBtn">
          <property name="toolTip">
           <string>Edit outbound</string>
          </property>
          <property name="text">
           <string>Edit Selection</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addRouteBtn">
          <property name="toolTip">
           <string>Add new route</string>
          </property>
          <property name="text">
           <string>Add Rule</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="delBtn">
          <property name="toolTip">
           <string>Delete outbound</string>
          </property>
          <property name="text">
           <string>Delete Selection</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QToolBox" name="toolBox">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>359</width>
        <height>536</height>
       </rect>
      </property>
      <attribute name="label">
       <string>General Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,4">
       <item>
        <widget class="QGroupBox" name="groupBox_7">
         <property name="title">
          <string>Misc Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Domain Strategy</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="domainStrategyCombo">
            <item>
             <property name="text">
              <string notr="true">AsIs</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">IPIfNonMatch</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string notr="true">IPOnDemand</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Default Outbound</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="defaultOutboundCombo">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Rule Order Settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Drag and drop to re-order the rules.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListWidget" name="ruleListWidget">
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="dragEnabled">
             <bool>true</bool>
            </property>
            <property name="dragDropMode">
             <enum>QAbstractItemView::InternalMove</enum>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="verticalScrollMode">
             <enum>QAbstractItemView::ScrollPerPixel</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
     <item>
      <widget class="QLabel" name="statusLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Status</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>addDefaultBtn</tabstop>
  <tabstop>editBtn</tabstop>
  <tabstop>addRouteBtn</tabstop>
  <tabstop>delBtn</tabstop>
  <tabstop>domainStrategyCombo</tabstop>
  <tabstop>defaultOutboundCombo</tabstop>
  <tabstop>ruleListWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RouteEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1115</x>
     <y>697</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RouteEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1115</x>
     <y>697</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
